### PR TITLE
BaSD-MS-05: Interviews - Components

### PR DESCRIPTION
**Pull request by ValentinaCollareta**
* Make a GET request to bring all the interviews &  create a table/list component to show them
* Make a POST & PUT request & create a form component with select & input components to show it
* Make a DELETE request to remove an specific interview & a modal component to cancel or confirm delete
* Create button components to cancel or confirm requests
* Follow screenflow for the components styling
* Show messages of success and failure related to every screen
* Validate the form data in both create and update request

### CODE REVIEW

**Comment by nicolasblasco** [comment]
@ValentinaCollareta Review of deployment;
- Problems to create an interview -> The server return a HTTP 400 Bad Request
- In the form, we could use a select instead of an input the status
- We should show a modal in a message with success or error
- We should refresh the page and close the modal when we delete an interview

**Comment by ValentinaCollareta** [comment]
Hola

**Comment by ValentinaCollareta** [comment]
@nicolasblasco I don't have any problem to create a new Interview ! 😕  Also, the page is refreshed after delete an interview ! For the rest, I'm working on it 👍

**Comment by nicolasblasco** [comment]
@ValentinaCollareta I tried again and it works! 
We should only focus on the other problems then!

**Review by nicolasblasco — APPROVED** [review]

File: .eslintrc.json [hunk 1]
```diff
@@ -1,27 +1,35 @@
...
+      "semi": [2, "always"],
+      "react/react-in-jsx-scope": "off",
+      "prettier/prettier": [
+          "error",
+          {
+            "endOfLine": "auto"
+          }
+        ],
+      "react/prop-types": 0,
+      "no-unused-vars": "off"
```
Comment (estefrare) [inline .eslintrc.json:33]:
```suggestion
      "no-unused-vars": "warn"
```

File: src/Components/Interviews/EditForm/index.js [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+          name="notes"
+          value={notesValue}
+          placeholder="Notes"
+          onChange={(e) => {
+            setNotesValue(e.target.value);
+          }}
+        />
+      </div>
+      <div className={styles.buttons}>
+        <button type="submit" className={styles.confirm} onClick={onSubmit}>
```
Comment (estefrare) [inline src/Components/Interviews/EditForm/index.js:241]:
```suggestion
        <button type="submit" className={styles.confirm}>
```

File: src/Components/Interviews/EditForm/index.js [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+      </div>
+      <div>
+        <label
+          style={{ color: 'red', width: '200px', textAlign: 'center', padding: '0px 20px' }}
+          htmlFor="messageStatus"
+        >
+          {errorStatus}
+        </label>
+        <label
+          style={{ color: 'red', width: '200px', textAlign: 'center', float: 'right' }}
```
Comment (estefrare) [inline src/Components/Interviews/EditForm/index.js:223]: DO NOT USE LINE STYLES

File: src/Components/Interviews/index.js [hunk 4]
```diff
@@ -1,10 +1,130 @@
...
+                  handleDelete(event, interview);
+                  setShowRemoveModal(true);
+                }}
+              />
+            </li>
+            <RemoveModal
+              show={showRemoveModal}
+              confirmRemoveModal={confirmRemoveModal}
+              closeRemoveModal={closeRemoveModal}
+            />
```
Comment (estefrare) [inline src/Components/Interviews/index.js:121]: this modal should be outside the map iteration

File: src/Components/Interviews/index.js [hunk 5]
```diff
@@ -1,10 +1,130 @@
...
+        setShowRemoveModal(false);
+        setShowSuccessDelete(true);
+      })
+      .then(() => {
+        setTimeout(function () {
+          if (setShowSuccessDelete) {
+            window.location.href = '/interviews';
+          }
+        }, 2000);
+      })
```
Comment (estefrare) [inline src/Components/Interviews/index.js:74]: refactor this

**Review by estefrare — CHANGES_REQUESTED** [review]

**Review by estefrare — APPROVED** [review]

**Review by nicolasblasco — APPROVED** [review]

**Pull request merged**